src/nvim/os/os_defs.h

```diff
@@ -1,10 +1,131 @@
 #ifndef NVIM_OS_OS_DEFS_H
 #define NVIM_OS_OS_DEFS_H
 
+#include <ctype.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
 #ifdef WIN32
 # include "nvim/os/win_defs.h"
 #else
 # include "nvim/os/unix_defs.h"
 #endif
 
+#ifdef HAVE_SYS_TIME_H
+# include <sys/time.h>
+#else
+# include <time.h>
+#endif
+
+#if defined(DIRSIZ) && !defined(MAXNAMLEN)
+# define MAXNAMLEN DIRSIZ
+#endif
+
+#if defined(NAME_MAX) && !defined(MAXNAMLEN)
+# define MAXNAMLEN NAME_MAX
+#endif
+
+// Default value.
+#ifndef MAXNAMLEN
+# define MAXNAMLEN 512
+#endif
+
+#define BASENAMELEN (MAXNAMLEN - 5)
```

Comment:
These checks for MAXNAMELEN, DIRSIZ, NAME_MAX and BASENAMELEN need to go into unix_defs.h - these are all Unix specific, Windows hash its own macro for this and win_defs.h needs to be able to set BASENAMELEN.